### PR TITLE
Add sound

### PR DESCRIPTION
**Pull request by wontheone1**
* sound effect
* BGM (Background music)

### CODE REVIEW

File: src/cljs/hello_world/game.cljs [hunk 1]
```diff
@@ -176,7 +176,9 @@
               bottom-button
               (fn []
                 (when (util/currently-playing-game?)
-                  (sound/play-beep! 300)
+                  (sound/play-beep! (sound/frequencies-in-major-scale-4th-octave
+                                      (mod (+ 1 util/row-col-num col)
+                                           (count sound/frequencies-in-major-scale-4th-octave))))
```
Comment (flyjwayur) [inline src/cljs/hello_world/game.cljs]: 1+5+0 =>  6 / 8 (sounds) = 0 ?
1 +5 + 1 => 7 / 8(sounds) = 0?? How dose it work? :)

Comment (wontheone1) [reply src/cljs/hello_world/game.cljs]:
```clojure
(mod (+ 1 5 0) 8)
=> 6
(mod (+ 1 5 1) 8)
=> 7
(mod (+ 1 5 2) 8)
=> 0
(mod (+ 1 5 3) 8)
=> 1
```
^ Is this enough example to understand how mod works?

Comment (wontheone1) [reply src/cljs/hello_world/game.cljs]: When first argument is negative
```clojure
(mod -6 8)
=> 2
(mod -7 8)
=> 1
(mod -8 8)
=> 0
```

Comment (flyjwayur) [reply src/cljs/hello_world/game.cljs]: Thanks a lot :) It's really helpful!

**Comment by wontheone1** [comment]
```clojure
(ns hello-world.sound-effect
  (:require [cljs-bach.synthesis
             :refer [add audio-context connect connect-> current-time destination
                     percussive gain sine square run-with]]
            [hello-world.util :as util]
            [leipzig.melody :as melody]
            ))

(enable-console-print!)

(def melody
  (melody/phrase [1 1 0.67 0.33 1 0.67 0.33 0.67 0.33 1]
                 [C5 C5 C5 D5 E5 E5 D5 E5 F5 G5]))

(println melody)
=>
({:pitch 523.26, :time 0, :duration 1} {:pitch 523.26, :time 1, :duration 1} {:pitch 523.26, :time 2, :duration 0.67} {:pitch 587.32, :time 2.67, :duration 0.33} {:pitch 659.26, :time 3, :duration 1} {:pitch 659.26, :time 4, :duration 0.67} {:pitch 587.32, :time 4.67, :duration 0.33} {:pitch 659.26, :time 5, :duration 0.67} {:pitch 698.46, :time 5.67, :duration 0.33} {:pitch 784, :time 6, :duration 1})
```
@flyjwayur this is how pitch key is made :), `melody/phrase` just makes sequence of maps with :pitch, :time, :duration fields

**Review by flyjwayur — COMMENTED** [review]
It's possible to print melody. I didnt think of it. It is really helpful. Thank u so much! :)

**Comment by wontheone1** [comment]
@flyjwayur Ideas to alternate 2 different melody.
1. use `then`
```clojure
(def composition1 ...)

(def composition2 ...)

(defn play-row-row-row-your-boat []
  (->> composition1
       (melody/then composition2)
       (melody/wherever (comp not :instrument) :instrument (melody/is bell))
       (play! context))
  (js/setTimeout play-row-row-row-your-boat 8000))

(play-row-row-row-your-boat)
```

**Comment by wontheone1** [comment]
Another idea is 
2. 
```Clojure
(def composition1 ...)

(def composition2 ...)

(def music [composition1 composition2])

(defn play-row-row-row-your-boat [index]
  (->> (music index)
       (melody/wherever (comp not :instrument) :instrument (melody/is bell))
       (play! context))
  (js/setTimeout (fn [] (play-row-row-row-your-boat (mod (inc index) (count music) 8000)) ; increase index but when played last phrase in vector, next index will be 0 because of "mod"

(play-row-row-row-your-boat)
```

**Comment by flyjwayur** [comment]
@wontheone1 
I tried with (play-row-row-row-your-boat 0), 0 for index.
But it didnt play comp1 & comp2. I dont the reason.

But the first solution works well :)

**Comment by flyjwayur** [comment]
@wontheone1 
The last commit seems it does not stop the music with audio on/off button.

**Comment by wontheone1** [comment]
@flyjwayur It does not stop sounds immediately but it stops after it's done playing current phrase.

**Comment by flyjwayur** [comment]
@wontheone1 Thanks 👍

**Pull request merged**